pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -453,6 +455,7 @@ public UnAckedMessageTracker getUnAckedMessageTracker() {
         try {
             return internalBatchReceiveAsync().get();
         } catch (InterruptedException | ExecutionException e) {
+            Thread.currentThread().interrupt();
```

Comment:
@eolivelli Can we interrupt current thread if ExecutionException occur?
@zjyingchang we also need do the same work in other implement of XXXConsumerImpl

Reply:
no, we have to call "Thread.currentThread().interrupt()" only while catching InterruptedException.

I suggest you to add some Util method and use it everywhere, this way we can still use multi catch

```
public static void handleInterruptedException(Throwable t) {
     if (t instanceof InterruptedException) {
           Thread.currentThread().interrupt();     
    }
}
```

Reply:
@eolivelli Please take a look again!  I add a Util class to handle InterruptedException and use it in XXXConsumeImpl.receive() and XXXConsumeImpl.receiveBatch() , or I need only add this Util method in ConsumeBase.